src/SpectatorView.Unity/Assets/SpatialAlignment/Scripts/Utilities/Extensions.cs

```diff
@@ -182,9 +182,9 @@ public static Task AsTask(this CancellationToken cancellationToken)
         /// <param name="task">The task to await.</param>
         /// <param name="cancellationToken">The cancellation token to stop awaiting.</param>
         /// <returns>The task that can be awaited unless the cancellation token is triggered.</returns>
-        public static Task Unless(this Task task, CancellationToken cancellationToken)
+        public static async Task Unless(this Task task, CancellationToken cancellationToken)
         {
-            return Task.WhenAny(task, cancellationToken.AsTask());
+            await (await Task.WhenAny(task, cancellationToken.AsTask()));
```

Comment:
> await (await Task.WhenAny(task, cancellationToken.AsTask())); [](start = 11, length = 62)

Actually this could be just one await and teh function not async, but the bug was that there was no await at all in old version.

Reply:
we tested this together locally and will leave this how it is.